numpy/core/src/multiarray/calculation.c

```diff
@@ -100,10 +100,10 @@ PyArray_ArgMax(PyArrayObject *op, int axis, PyArrayObject *out)
     }
 
     if (!out) {
-        rp = (PyArrayObject *)PyArray_New(Py_TYPE(ap), PyArray_NDIM(ap)-1,
-                                          PyArray_DIMS(ap), NPY_INTP,
-                                          NULL, NULL, 0, 0,
-                                          (PyObject *)ap);
+        rp = (PyArrayObject *)PyArray_NewFromDescr(
+                Py_TYPE(ap), PyArray_DescrFromType(NPY_INTP),
```

Comment:
We're making the assumption that `PyArray_DescrFromType` does not return `NULL` here. This is already assumed elsewhere

Reply:
I assume this is safe because it is a known numeric type?

Reply:
Yep, `PyArray_DescrFromType` starts with an error-free lookup of builtin type numbers